From ff1b15e1a65f6a881d9f9659559e2d6958d1b1fb Mon Sep 17 00:00:00 2001
From: Elaine Zhang <zhangqing@rock-chips.com>
Date: Mon, 24 Aug 2020 15:16:13 +0800
Subject: [PATCH] net: can: rockchip: modify the usleep to udelay

usleep cannot be used in interrupts,
use udelay instead of usleep.

Generally, the reset time is 5 cycles(PCLK).

Signed-off-by: Elaine Zhang <zhangqing@rock-chips.com>
Change-Id: I7cf1cac018aa2b440ed79dcb4e8fb77607930bb0
---
 drivers/net/can/rockchip/rockchip_can.c   | 2 +-
 drivers/net/can/rockchip/rockchip_canfd.c | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/net/can/rockchip/rockchip_can.c b/drivers/net/can/rockchip/rockchip_can.c
index 6dfb741d7128..461c6d5cb66c 100644
--- a/drivers/net/can/rockchip/rockchip_can.c
+++ b/drivers/net/can/rockchip/rockchip_can.c
@@ -134,7 +134,7 @@ static int set_reset_mode(struct net_device *ndev)
 	struct rockchip_can *rcan = netdev_priv(ndev);
 
 	reset_control_assert(rcan->reset);
-	usleep_range(2000, 2500);
+	udelay(2);
 	reset_control_deassert(rcan->reset);
 
 	writel(0, rcan->base + CAN_MODE);
diff --git a/drivers/net/can/rockchip/rockchip_canfd.c b/drivers/net/can/rockchip/rockchip_canfd.c
index 03f7a30c2f01..071bb4e16f1b 100644
--- a/drivers/net/can/rockchip/rockchip_canfd.c
+++ b/drivers/net/can/rockchip/rockchip_canfd.c
@@ -254,7 +254,7 @@ static int set_reset_mode(struct net_device *ndev)
 	struct rockchip_canfd *rcan = netdev_priv(ndev);
 
 	reset_control_assert(rcan->reset);
-	usleep_range(2000, 2500);/* estimated value */
+	udelay(2);
 	reset_control_deassert(rcan->reset);
 
 	rockchip_canfd_write(rcan, CAN_MODE, 0);
-- 
2.35.3

